pkg/agent/run.go

```diff
@@ -105,21 +114,25 @@ func Run(cmd *cobra.Command, args []string) {
 
 		err := http.ListenAndServe(":8081", server)
 		if err != nil && !errors.Is(err, http.ErrServerClosed) {
-			logs.Log.Fatalf("failed to run the health check server: %s", err)
+			return fmt.Errorf("failed to run the health check server: %s", err)
 		}
-	}()
+		// The agent must stop if the management server stops
+		cancel()
+		return nil
+	})
```

Comment:
In a future PR we should add a signal handler to catch sigterm (ctrl-c) and cancel the parent context, and add another Go routine which calls `server.Stop` after the context is cancelled.

As it stands, sigterm will cause the process (all threads) to immediately  exit without any graceful shutdown.
 * https://pkg.go.dev/os/signal#hdr-Default_behavior_of_signals_in_Go_programs